### PR TITLE
Add length to FileAppender to avoid a call to S3 when writing.

### PR DESCRIPTION
**Pull request by rdblue**

### CODE REVIEW

File: core/src/main/java/com/netflix/iceberg/avro/AvroFileAppender.java [hunk 1]
```diff
@@ -56,6 +59,18 @@ public Metrics metrics() {
     return new Metrics(numRecords, null, null, null);
   }
 
+  @Override
+  public long length() {
+    if (stream != null) {
```
Comment (rdsr) [inline core/src/main/java/com/netflix/iceberg/avro/AvroFileAppender.java]: The API make the contract that this is valid only when the file is closed. Would it make sense to check that in Avro and Orc file appenders?

Comment (rdblue) [reply core/src/main/java/com/netflix/iceberg/avro/AvroFileAppender.java]: Yes.

Comment (rdblue) [reply core/src/main/java/com/netflix/iceberg/avro/AvroFileAppender.java]: Fixed. Please have another look.

**Comment by rdblue** [comment]
@rdsr, I rebased and fixed the issue you caught. Thanks!

**Pull request merged**